hibernate-ehcache/src/main/java/org/hibernate/cache/ehcache/EhCacheRegionFactory.java

```diff
@@ -106,4 +109,14 @@ public void stop() {
 		}
 	}
 
+	static void setCacheManagerNameIfNeeded(SessionFactoryOptions settings, Configuration configuration) {
```

Comment:
Longer term I wonder if we should instead pass in the SessionFactory reference.  SessionFactory already defines a UUID...

Reply:
Yeah, that'd be great… I saw the name could be `null` in the `SessionFactoryOptions`, so… free styled. But if the `SessionFactory` _always_ named, it be nicer indeed.

Reply:
It would be great.  The problem though is that Hibernate itself does not always have good visibility into a good name.  A "container" (WidFly, Spring, etc) has a much better idea of that.  Wildfly, e.g., does in deed set the SF name if the user has not.

It's hard to enforce that that always happen uniquely though.

However, there is always a UUID : `org.hibernate.engine.spi.SessionFactoryImplementor#getUuid`